test/lib/secretFactory.test.js

```diff
@@ -163,6 +163,22 @@ describe('Secret Factory', () => {
                     });
                 })
         );
+
+        it('skip unseal when secret not found', () => {
+            datastore.get.withArgs({
+                table: 'secrets',
+                params: {
+                    id
+                }
+            }).resolves(null);
+
+            return factory.get({ pipelineId, name })
+                .then(model => {
+                    assert.isTrue(datastore.get.calledOnce);
+                    assert.notCalled(ironMock.unseal);
+                    assert.isNull(model);
```

Comment:
That's an odd assertion. Are you checking for `null` because your mock returns null? 

If you're looking to see if data came back from your mock, it would be more deterministic if you had your mock return something that only could come back from the mock.